src/main/java/webserver/controllers/TodoController.java

```diff
@@ -98,10 +99,70 @@ public TodoController(TodoService todoService) {
             .build();
     };
 
+    public Callback<Request, String> getTodoDetailEdit = (request) -> {
+        final int id = getTodoIdFromPath(request.getPath());
+        String html = "";
+        int responseCode = 0;
+
+        Mustache template = MustacheUtil.getTemplate("todo-item-edit.mustache");
+        List<Todo> allTodos = todoService.getTodos();
+        Todo todo = TodosHelper.getTodo(id, allTodos);
```

Comment:
This could be a good place to introduce a method to your service/interface for TODO items.  Something like 

```java
interface TodoService {
  Todo getTodo(int id);
}
```

The implementation will probably have the same two lines of code in it, but there's real power in expressing _exactly_ what you want when looking up data.  It can also make mocking/stubbing easier in your tests.

This comes up in data layers a lot, where the database might be querying way more records than necessary (looking up all todo items, for example) only for the application to sort through them and throw most of those results out.